src/sentry/integrations/aws_lambda/integration.py

```diff
@@ -347,19 +350,30 @@ def dispatch(self, request, pipeline):
         lambda_functions = get_supported_functions(lambda_client)
         lambda_functions.sort(key=lambda x: x["FunctionName"].lower())
 
-        failures = []
-        success_count = 0
-
-        for function in lambda_functions:
+        def is_lambda_enabled(function):
             name = function["FunctionName"]
             # check to see if the user wants to enable this function
-            if not enabled_lambdas.get(name):
-                continue
+            return enabled_lambdas.get(name)
+
+        lambda_functions = filter(is_lambda_enabled, lambda_functions)
 
+        def _enable_lambda(function):
             try:
                 enable_single_lambda(lambda_client, function, sentry_project_dsn)
-                success_count += 1
+                return (True, function, None)
             except Exception as e:
+                return (False, function, e)
+
+        failures = []
+        success_count = 0
+
+        # use threading here to parallelize requests
+        results = list(_lambda_setup_thread_pool.map(_enable_lambda, lambda_functions))
```

Comment:
I would inline this into the for loop actually, so you can process them as they complete

Reply:
Should we add a timeout here as well? It's possible this code could deadlock (though the enable_single_lambda may already have some kind of timeout)

Reply:
@EvanPurkhiser  timeout added